sdk/src/test/java/io/opentelemetry/sdk/trace/SpanBuilderSdkTest.java

```diff
@@ -177,6 +177,24 @@ public void changingAttributes_NoEffectAfterAddLink() {
     }
   }
 
+  @Test
+  public void addLink_NoEffectAfterStartSpan() {
+    Span.Builder spanBuilder = tracerSdk.spanBuilder(SPAN_NAME);
+    spanBuilder.addLink(sampledSpanContext);
+    RecordEventsReadableSpan span = (RecordEventsReadableSpan) spanBuilder.startSpan();
+    try {
+      assertThat(span.toSpanData().getLinks())
+          .containsExactly(
+              Link.create(sampledSpanContext, Collections.<String, AttributeValue>emptyMap()));
+      spanBuilder.addLink(sampledSpanContext);
```

Comment:
You might want to use a different SpanContext here, otherwise the missing effect could also come from a check that links are not duplicated (or a set being used internally).

Reply:
Done, also added a comment to explain.